src/autoscaler/pruner/pruner.go

```diff
@@ -0,0 +1,57 @@
+package pruner
+
+import (
+	"time"
+
+	"code.cloudfoundry.org/clock"
+	"code.cloudfoundry.org/lager"
+)
+
+type Pruner interface {
+	PruneOldData()
+}
+
+type DbPrunerRunner struct {
+	pruner   Pruner
+	name     string
+	interval time.Duration
+	clock    clock.Clock
+	doneChan chan bool
+	logger   lager.Logger
+}
+
+func NewDbPrunerRunner(pruner Pruner, name string, interval time.Duration, clock clock.Clock, logger lager.Logger) *DbPrunerRunner {
+	return &DbPrunerRunner{
+		pruner:   pruner,
+		name:     name,
+		interval: interval,
+		clock:    clock,
+		logger:   logger,
+		doneChan: make(chan bool),
+	}
+}
+
+func (dpr *DbPrunerRunner) Start() {
```

Comment:
Why do we need Start/Stop if it gets wrapped in an ifrit.Process?  We just need a Pruner() or Run() or whatever.  The rest is overkill.